source/extensions/transport_sockets/http_11_proxy/connect.cc

```diff
@@ -67,27 +68,34 @@ Network::IoResult UpstreamHttp11ConnectSocket::doRead(Buffer::Instance& buffer)
       return {Network::PostIoAction::Close, 0, false};
     }
     absl::string_view peek_data(peek_buf, result.return_value_);
-    size_t index = peek_data.find("\r\n\r\n");
-    if (index == absl::string_view::npos) {
-      if (result.return_value_ == MAX_RESPONSE_HEADER_SIZE) {
-        ENVOY_CONN_LOG(trace, "failed to receive CONNECT headers within {} bytes",
-                       callbacks_->connection(), MAX_RESPONSE_HEADER_SIZE);
-        return {Network::PostIoAction::Close, 0, false};
-      }
-      return Network::IoResult{Network::PostIoAction::KeepOpen, 0, false};
+    unsigned long len = 0;
+    bool error = false;
+    bool is_valid_connect_response = isValidConnectResponse(peek_data, error, len);
+
+    if (error) {
+      ENVOY_CONN_LOG(trace, "Response does not appear to be a successful CONNECT upgrade",
+                     callbacks_->connection());
+      return {Network::PostIoAction::Close, 0, false};
     }
-    result = callbacks_->ioHandle().read(buffer, index + 4);
-    if (!result.ok() || result.return_value_ != index + 4) {
-      ENVOY_CONN_LOG(trace, "failed to drain CONNECT header", callbacks_->connection());
+    if (!is_valid_connect_response && len == 200) {
```

Comment:
Should this be MAX_RESPONSE_HEADER_SIZE instead of 200? (I thought 200 meant the status code initially) For that matter, should it be >= MAX_RESPONSE_HEADER_SIZE? (I confess I'm not quite sure what the